# --- T2-COPYRIGHT-NOTE-BEGIN ---
# T2 SDE: package/*/afbinit/hotfix-mmap.patch
# Copyright (C) 2022 The T2 SDE Project
# 
# This Copyright note is generated by scripts/Create-CopyPatch,
# more information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License version 2 as used by the T2 SDE.
# --- T2-COPYRIGHT-NOTE-END ---

--- ./afbinit.c.vanilla
+++ ./afbinit.c
@@ -236,7 +236,7 @@
 	/* MMAP the registers. */
 	uregs = mmap(0, 0x2000,
 		     PROT_READ | PROT_WRITE,
-		     MAP_PRIVATE,
+		     MAP_SHARED,
 		     afb_fd,
 		     0x04000000);
 	if (uregs == (void *)-1L) {
@@ -246,7 +246,7 @@
 
 	kregs = mmap(0, 0x2000,
 		     PROT_READ | PROT_WRITE,
-		     MAP_PRIVATE,
+		     MAP_SHARED,
 		     afb_fd,
 		     0x0bc04000);
 	if (kregs == (void *)-1L) {
